app/controllers/api/groups_controller.rb

```diff
@@ -80,5 +80,54 @@ def include_students(fields)
       end
     end
 
+    # Update the group's marks for the given assignment.
+    def update_marks
+      assignment = Assignment.find_by_id(params[:assignment_id])
+      if assignment.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+          'No assignment exists with that id' }, status: 404
+        return
+      end
+
+      group = Group.find_by_id(params[:id])
+      if group.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+          'No group exists with that id' }, status: 404
+        return
+      elsif group.submissions.first.nil?
```

Comment:
We really want to be using the `grouping` properties `has_submission?` and `current_submission_used` here. This code may be incorrect when there are multiple submissions.